src/core/tsi/s2a/record_protocol/s2a_crypter.cc

```diff
@@ -357,10 +398,12 @@ void check_half_connection(s2a_crypter* crypter, bool in_half_connection,
     GPR_ASSERT(half_connection->traffic_secret[i] ==
                expected_traffic_secret[i]);
   }
-  GPR_ASSERT(half_connection->fixed_nonce != nullptr);
-  GPR_ASSERT(half_connection->fixed_nonce_size == expected_fixed_nonce_size);
-  for (size_t i = 0; i < expected_fixed_nonce_size; i++) {
-    GPR_ASSERT(half_connection->fixed_nonce[i] == expected_fixed_nonce[i]);
+  if (verify_nonce) {
```

Comment:
What would be the case where we don't want to verify the nonce?

Reply:
I've tried to have 2 classes of unit tests related to half connections: some which verify the traffic secret to key/nonce, and others which verify traffic secret to advanced traffic secret. The `verify_nonce` variable is set to true in the latter case.